rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -2464,11 +2467,17 @@ rmw_send_request(
   // and any number.
   opts.consolidation = z_query_consolidation_latest();
   opts.value.payload = z_bytes_t{data_length, reinterpret_cast<const uint8_t *>(request_bytes)};
-  client_data->zn_closure_reply =
+  z_owned_closure_reply_t zn_closure_reply =
```

Comment:
Shouldn't we check that `is_shutdown_` is `false` before issuing new queries for this client ? (Maybe there is another check somewhere else that already prevents that I'm not aware of)

Reply:
Yes, good point.  I'll add this in as well.